include/envoy/api/io_error.h

```diff
@@ -77,11 +51,20 @@ template <typename T> struct IoCallResult {
 
   virtual ~IoCallResult() {}
 
+  IoCallResult& operator=(IoCallResult&& result) {
+    rc_ = result.rc_;
+    err_ = std::move(result.err_);
+    return *this;
+  }
+
   T rc_;
   IoErrorPtr err_;
 };
 
 using IoCallUintResult = IoCallResult<uint64_t>;
 
+#define IO_CALL_RESULT_NO_ERROR                                                                    \
```

Comment:
nit: can this just be an inline free function? Does it need to be a macro?

Reply:
done